### DISCUSSION TITLE
Monitor Interface randomly Drops everything

### DISCUSSION THREAD

**Discussion by ccpaasmyser** [Security-Onion-Solutions/securityonion#8635] [Unsupported Versions]
Our security onion environment runs really smoothly. However, the monitor interface randomly stops getting inbound traffic and the Inbound drops in the grid view increase from nothing to, I'm guessing, everything.  I'm not really sure where to look for logs to try to figure out what's going on.  I've noticed this the past couple days, and it seems a reboot of the system get it back up and running.

I've attached a screenshot of the grafana. You can see in the Monitor Traffic where it stopped and after a bit of time i rebooted it and it started again.
![Untitled](https://user-images.githubusercontent.com/100237680/187519890-c89da0d9-3c05-4ca3-aa69-a62156f182bf.png)

**Comment by dougburks** [comment]
What kind of NIC?

Have you checked for errors on the NIC?

Have you checked your upstream tap or span port for errors?

**Reply by ccpaasmyser** [reply]
I haven't been able to check that port specifically to make sure its still sending data, however, the Monitor Interface Drops section seems to follow the amount of data that the machine should be getting.  I'm keeping an eye on it right now and will have my network admin verify that the next time it drops.

I'm checking with the hardware guy and will get back on the NIC.

Is there a log somewhere within security onion that i can see errors on the NIC, or somewhere within CentOS7?

**Reply by ccpaasmyser** [reply]
Ethernet ports are both 10Gb ports (Intel X550 Dual Port 10GbE BASE-T)

System is still running with monitor traffic so i'm waiting for it to fail.

**Reply by ccpaasmyser** [reply]
this is the current ifconfig, which shows port errors. I'll keep an eye on this and once it fails post again.

```
[root@securityonion ~]# ifconfig
bond0: flags=5443<UP,BROADCAST,RUNNING,PROMISC,MASTER,MULTICAST>  mtu 1500
        ether 78:ac:44:64:82:81  txqueuelen 1000  (Ethernet)
        RX packets 1957952436  bytes 1442378213815 (1.3 TiB)
        RX errors 2  dropped 5642  overruns 0  frame 2
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.255.0  broadcast 172.17.0.255
        ether 02:42:e5:4a:7a:fd  txqueuelen 0  (Ethernet)
        RX packets 61644341  bytes 382565355117 (356.2 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 67821180  bytes 383421411963 (357.0 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

em1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.16.24.191  netmask 255.255.248.0  broadcast 172.16.31.255
        ether 78:ac:44:64:82:80  txqueuelen 1000  (Ethernet)
        RX packets 495142  bytes 41830134 (39.8 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 88602  bytes 56502179 (53.8 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

em2: flags=2499<UP,BROADCAST,RUNNING,NOARP,PROMISC,SLAVE>  mtu 1500
        ether 78:ac:44:64:82:81  txqueuelen 1000  (Ethernet)
        RX packets 1957953335  bytes 1442378476112 (1.3 TiB)
        RX errors 4  dropped 5486  overruns 0  frame 4
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

**Reply by dougburks** [reply]
You might also check `dmesg` for possible clues.

**Reply by ccpaasmyser** [reply]
Just getting in today. I'm noticing the Swap is at 82% usage, which is weird since the system barely ever uses swap, so not sure how that got so high overnight.

Not sure if it helps now as the machine is still technically running but i pulled off a dmesg and attached it.
[dmesg.txt](https://github.com/Security-Onion-Solutions/securityonion/files/9469585/dmesg.txt)


I also pulled another ifconfig and it shows the below, still not a ton of errors as far as i can tell.
```
[root@securityonion ~]# ifconfig
bond0: flags=5443<UP,BROADCAST,RUNNING,PROMISC,MASTER,MULTICAST>  mtu 1500
        ether 78:ac:44:64:82:81  txqueuelen 1000  (Ethernet)
        RX packets 5779473855  bytes 4496709004568 (4.0 TiB)
        RX errors 2  dropped 23271  overruns 0  frame 2
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.255.0  broadcast 172.17.0.255
        ether 02:42:e5:4a:7a:fd  txqueuelen 0  (Ethernet)
        RX packets 217985586  bytes 1359698456145 (1.2 TiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 239242630  bytes 1362480878097 (1.2 TiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

em1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.16.24.191  netmask 255.255.248.0  broadcast 172.16.31.255
        ether 78:ac:44:64:82:80  txqueuelen 1000  (Ethernet)
        RX packets 1853792  bytes 191110797 (182.2 MiB)
        RX errors 0  dropped 4  overruns 0  frame 0
        TX packets 302900  bytes 120705735 (115.1 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

em2: flags=2499<UP,BROADCAST,RUNNING,NOARP,PROMISC,SLAVE>  mtu 1500
        ether 78:ac:44:64:82:81  txqueuelen 1000  (Ethernet)
        RX packets 5779474557  bytes 4496709116429 (4.0 TiB)
        RX errors 4  dropped 23115  overruns 0  frame 4
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

**Reply by ccpaasmyser** [reply]
swap is still climbing up to 95% while the memory usage graph is sitting at 36.6%

**Reply by ccpaasmyser** [reply]
Swap has successfully gone up to 100% however, the entire system seems to be working fine. still no drops on the monitor traffic...

**Reply by dougburks** [reply]
If RAM and swap hit 100%, then the OS will start killing processes. You may need to increase RAM.

**Reply by ccpaasmyser** [reply]
Even if the memory usage is still around 36%? How much swap should be on the system? I'm pretty sure we maxed out the server at 128gb of ram.

**Reply by ccpaasmyser** [reply]
For instance this is how it currently sits. The system seems to be functioning fine. Just not sure if the swap usage can be resolved. I only made the swap partition 4Gbs as thats what the OS made it by default. Didn't think it would need to be any bigger.
![Untitled](https://user-images.githubusercontent.com/100237680/188221376-f7d630c4-a383-424a-b467-a1501cb06555.png)

**Reply by Acewiza** [reply]
Not knowing your traffic patterns or really any other factors that might influence the overall workload, I wonder why you need swap at all with 128G RAM?

**Reply by isaacgolding** [reply]
For what it's worth you can disable the swap until next reboot by typing run the command **sudo swapoff -a**
I'd suggest doing this and then start monitoring the system logs over the next couple days to see if you have any errors showing up.

**Reply by ccpaasmyser** [reply]
Honestly the swap issue isn't really seemingly to cause any issues, However. Today is the first day since last week, or the week before that the monitor traffic has completely dropped.  I had my network people check the mirror span port and it is still up without errors. an ifconfig shows the following. The em2 interface is the monitoring port

```

[root@securityonion ~]# sudo ifconfig
bond0: flags=5443<UP,BROADCAST,RUNNING,PROMISC,MASTER,MULTICAST>  mtu 1500
        ether 78:ac:44:64:82:81  txqueuelen 1000  (Ethernet)
        RX packets 364463719  bytes 260005812206 (242.1 GiB)
        RX errors 0  dropped 36648171  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.255.0  broadcast 172.17.0.255
        ether 02:42:12:88:98:e6  txqueuelen 0  (Ethernet)
        RX packets 11060749  bytes 70085784988 (65.2 GiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 11732136  bytes 70174368365 (65.3 GiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

em1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.16.24.191  netmask 255.255.248.0  broadcast 172.16.31.255
        ether 78:ac:44:64:82:80  txqueuelen 1000  (Ethernet)
        RX packets 120387  bytes 12506124 (11.9 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 25830  bytes 23976081 (22.8 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

em2: flags=2499<UP,BROADCAST,RUNNING,NOARP,PROMISC,SLAVE>  mtu 1500
        ether 78:ac:44:64:82:81  txqueuelen 1000  (Ethernet)
        RX packets 364463897  bytes 260005827746 (242.1 GiB)
        RX errors 3  dropped 36648015  overruns 0  frame 3
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

```

**Reply by Acewiza** [reply]
Whatever it is, ur swap indicator isn't right.  Spending CPU cycles to manage a full swap seems a very likely way to drop packets, to me.

**Reply by Acewiza** [reply]
FWIW, my swap runs around 16% min/max/avg but the indicator always spikes hard when I launch Grafana.  SO guys might wanna take a look at that widget.  In any case, it ain't right.

**Reply by dougburks** [reply]
> my swap runs around 16% min/max/avg but the indicator always spikes hard when I launch Grafana. SO guys might wanna take a look at that widget. In any case, it ain't right.

@Acewiza I checked a few deployments and they are not exhibiting this behavior. The widget seems to be working correctly for me. If the indicator spikes hard when you launch Grafana, then your system may be low on RAM. If you'd like to further discuss your system, please start a new discussion to avoid confusion with the original poster.

**Reply by Acewiza** [reply]
No, it works fine on my onion.  Just curious what's causing the OP's problem - rather unusual.

**Reply by ccpaasmyser** [reply]
So i was away for a couple days and came back to all drops in the monitor traffic so i rebooted and now its not coming back after a reboot, which is unusual, it usuall resolves itself after a reboot. Not really sure where to gofrom here or where to look.

**Reply by dougburks** [reply]
If you run tcpdump on the sniffing interface, do you see any traffic?

If your sniffing interface is connected to a span port, then it's possible that the switch reaches capacity and stops copying traffic to the span port until it is back to a lower load. This is one of the reasons we recommend dedicated taps where possible:
https://docs.securityonion.net/en/2.3/hardware.html#packets

**Reply by ccpaasmyser** [reply]
i do not see any traffic on the tcpdump of interface em2 which is my monitoring interface. I had my network guy shut off the mirror and start it back up and nothing came through at all. Still no luck. I see tons of drops on the ifconfig for em2 interface though.

~~~
[root@securityonion ~]# ifconfig
bond0: flags=5443<UP,BROADCAST,RUNNING,PROMISC,MASTER,MULTICAST>  mtu 1500
        ether 78:ac:44:64:82:81  txqueuelen 1000  (Ethernet)
        RX packets 3187012  bytes 2278120663 (2.1 GiB)
        RX errors 118  dropped 36480331452  overruns 0  frame 118
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.255.0  broadcast 172.17.0.255
        ether 02:42:0e:30:d1:23  txqueuelen 0  (Ethernet)
        RX packets 129038849  bytes 1638274759403 (1.4 TiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 132108686  bytes 1632041397279 (1.4 TiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

em1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.16.24.191  netmask 255.255.248.0  broadcast 172.16.31.255
        ether 78:ac:44:64:82:80  txqueuelen 1000  (Ethernet)
        RX packets 10120584  bytes 1015628904 (968.5 MiB)
        RX errors 2  dropped 28  overruns 0  frame 2
        TX packets 692372  bytes 124358335 (118.5 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

em2: flags=2499<UP,BROADCAST,RUNNING,NOARP,PROMISC,SLAVE>  mtu 1500
        ether 78:ac:44:64:82:81  txqueuelen 1000  (Ethernet)
        RX packets 3187725  bytes 2278178923 (2.1 GiB)
        RX errors 120  dropped 36480331437  overruns 0  frame 120
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
~~~

**Reply by ccpaasmyser** [reply]
I'd just like to point out that our old system used this same port without problem. We didn't change anything with the span port between the old system and the new one.

**Reply by dougburks** [reply]
Could be a bad NIC or bad cable.

**Reply by ccpaasmyser** [reply]
i guess i can try another cable and or nic. How do i change what my monitoring interface is?

**Reply by ccpaasmyser** [reply]
i switched cables and monitor nics and reran sosetup to get it all setup right. and logs are now coming in. we will see how the swap and drops progress

**Reply by ccpaasmyser** [reply]
The swap has climbed back to 100% usage, the system seems to still be running fine. Not drops in the monitoring interface though.
